drivers/spi/spi_sam0.c

```diff
@@ -129,18 +133,36 @@ static int spi_sam0_configure(const struct device *dev,
 	div = (SOC_ATMEL_SAM0_GCLK0_FREQ_HZ / config->frequency) / 2U - 1;
 	div = CLAMP(div, 0, UINT8_MAX);
 
+#ifdef SERCOM_SPI_CTRLC_MASK
+	/* LENGTH.LEN must only be enabled when CTRLC.bit.DATA32B is enabled.
+	 * Since we are about to explicitly disable it, we need to clear the LENGTH register.
+	 */
+	length.reg = SERCOM_SPI_LENGTH_RESETVALUE;
+
+	/* Disable inter-character spacing and the 32-bit read/write extension */
+	ctrlc.reg = SERCOM_SPI_CTRLC_RESETVALUE;
+#endif
+
 	/* Update the configuration only if it has changed */
-	if (regs->CTRLA.reg != ctrla.reg || regs->CTRLB.reg != ctrlb.reg ||
-	    regs->BAUD.reg != div) {
+	if (regs->CTRLA.reg != ctrla.reg || regs->CTRLB.reg != ctrlb.reg || regs->BAUD.reg != div
+#ifdef SERCOM_SPI_CTRLC_MASK
+		|| regs->LENGTH.reg != length.reg || regs->CTRLC.reg != ctrlc.reg
+#endif
+	) {
```

Comment:
The first commit has an issue with the opening `(` bracket, which is fixed in the second commit. This results in a non-bisect able history, please fix the first commit.

Reply:
My apologies, very sloppy of me.

Reply:
@pdgendt Done!